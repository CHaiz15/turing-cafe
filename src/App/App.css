@import url('https://fonts.googleapis.com/css?family=Pacifico');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url('../assets/cafe-img.jpeg');
  width: 100%;
  height: 100vh;
  overflow: scroll;
  background-repeat: no-repeat;
  background-size: cover;
  text-align: center;
}

.app-title {
  margin: 1% 0 0 0;
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}