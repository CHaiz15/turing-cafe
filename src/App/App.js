import React, { Component } from 'react';
import './App.css';
import Reservations from '../Reservations/Reservations'
import Form from '../Form/Form'


class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: []
    }
  }

  addReservation = (newRes) => {
    const options = {
      method: 'POST',
      body: JSON.stringify({
        ...newRes
      }),
      headers: {
        'Content-Type': 'application/json'
      }
    }
      fetch('http://localhost:3001/api/v1/reservations', options)
      .then(response => response.json())
      .then(reservation => this.setState({reservations: [...this.state.reservations, reservation]}))
      .catch(error => console.log(error))
  }

  deleteReservation = (id) => {
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    }
      fetch('http://localhost:3001/api/v1/reservations/' + id, options)
      .then(response => response.json())
      .then(reservations => this.setState({reservations}))
      .catch(error => console.log(error))
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
    .then(response => response.json())
    .then(reservations => this.setState({reservations}))
    .catch(error => console.log(error))
  }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form addReservation={this.addReservation} />
        </div>
        <div className='resy-container'>
          <Reservations deleteReservation={this.deleteReservation} reservations={this.state.reservations} />
        </div>
      </div>
    )
  }
}

export default App;
