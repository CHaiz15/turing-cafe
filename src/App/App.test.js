import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should be an instance of an app componenet', () => {
  let wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
})

it('should change state when addReservation is called', () => {
  const wrapper = shallow(<App />);
  const mockNewRes = {
      name: 'Richie',
      date: '02/12',
      time: '7:00',
      number: 1
    }
  wrapper.instance().addReservation(mockNewRes)
  expect(wrapper.state('reservations')).toEqual([mockNewRes])
})
