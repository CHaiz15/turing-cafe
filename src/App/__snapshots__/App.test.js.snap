// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be an instance of an app componenet 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <Reservations
      deleteReservation={[Function]}
      reservations={Array []}
    />
  </div>
</div>
`;
